code/modules/vr/vr_avatar.dm

```diff
@@ -16,72 +17,120 @@
 	..()
 
 /mob/living/carbon/human/virtual_reality/death()
-	return_to_lobby()
+	myroom.players.Remove(src)
+	vr_all_players.Remove(src)
+	if((myroom.players.len == 0) && (myroom.expires == 1))
```

Comment:
`if((!myroom.players.len) && (myroom.expires))` That is, unless you expect len to be negative or expires to be negative